nifi-nar-bundles/nifi-cdc/nifi-cdc-mssql-bundle/nifi-cdc-mssql-processors/src/main/java/org/apache/nifi/cdc/mssql/processors/CaptureChangeMSSQL.java

```diff
@@ -0,0 +1,409 @@
+    protected Map<String,String> maxValueProperties;
+    protected MSSQLCDCUtils mssqlcdcUtils = new MSSQLCDCUtils();
+
+    public MSSQLCDCUtils getMssqlcdcUtils(){
+        return mssqlcdcUtils;
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(RECORD_WRITER);
+        descriptors.add(DBCP_SERVICE);
+        descriptors.add(CDC_TABLES);
+        descriptors.add(TAKE_INITIAL_SNAPSHOT);
+        descriptors.add(FULL_SNAPSHOT_ROW_LIMIT);
+        descriptors.add(INCLUDE_PREUPDATE_VALUES);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_SUCCESS);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        if(!propertyDescriptorName.startsWith("initial.timestamp.")){
+            return null;
+        }
+
+        return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .displayName(propertyDescriptorName)
+                .required(false)
+                .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(AttributeExpression.ResultType.STRING, true))
+                .addValidator(StandardValidators.ATTRIBUTE_KEY_PROPERTY_NAME_VALIDATOR)
+                .expressionLanguageSupported(false)
+                .dynamic(true)
+                .build();
+    }
+
+    @Override
+    public void onTrigger(ProcessContext processContext, ProcessSessionFactory processSessionFactory) throws ProcessException {
+        ProcessSession session = processSessionFactory.createSession();
+
+        final ComponentLog logger = getLogger();
+        final RecordSetWriterFactory writerFactory = processContext.getProperty(RECORD_WRITER).asControllerService(RecordSetWriterFactory.class);
+        final DBCPService dbcpService = processContext.getProperty(DBCP_SERVICE).asControllerService(DBCPService.class);
+
+        final boolean takeInitialSnapshot = processContext.getProperty(TAKE_INITIAL_SNAPSHOT).asBoolean();
+        final boolean includePreupdateValues = processContext.getProperty(INCLUDE_PREUPDATE_VALUES).asBoolean();
+        final int fullSnapshotRowLimit = processContext.getProperty(FULL_SNAPSHOT_ROW_LIMIT).evaluateAttributeExpressions().asInteger();
+
+        final String[] allTables = schemaCache.keySet().toArray(new String[schemaCache.size()]);
+
+        String[] tables = StringUtils
+                .split(processContext.getProperty(CDC_TABLES).evaluateAttributeExpressions().getValue(), ",");
```

Comment:
To the other comment, this might be better handled with a stream split -> trim -> collect in order to handle whitespace between the delimiter and table names

Reply:
@mattyb149 I've updated the code, and tweaked a unit test to cover it.  What are your thoughts on working towards merging this :) I feel like it's gotten quite a bit of external testing by users at this point.

Reply:
@mattyb149 @patricker We still trying to get this merged into master? Anything I can do from the testing standpoint to help that along?